cloud/terraform/terraform_controller.py

```diff
@@ -0,0 +1,122 @@
+import os
+
+
+class TerraformController:
+    def __init__(self, cloud):
+        self.cloud = cloud
+
+        #TODO: where is this executed from? change path?
+        if os.path.exists("main.tf"):
+            os.remove("main.tf")
+        header = """
+terraform {
+    required_providers {
+        aws = {
+        source  = "hashicorp/aws"
+        version = "~> 3.27"
+        }
+    }
+
+    required_version = ">= 0.14.9"
+}
+"""
+        self._add_to_config_file(header)
+
+    def _add_to_config_file(self, addition):
+        with open("main.tf", "a") as file:
+            file.write(addition)
+        
+    def _string_in_config(self, test_string):
+        with open("main.tf", "r") as file:
+            return (test_string in file.read())
+
+    # Credentials can be passed to terraform by adding the to the provider block
+    def aws_credentials(self, access_key, secret_key, token):
+        """
+        Give terraform controller aws credentials used on the configuration
```

Comment:
Open to discussion: Are docstrings really needed?

IMO, function name and code inside should be self-explanatory enough to avoid having docstrings in every function.